datafusion/core/tests/physical_optimizer/partition_statistics.rs

```diff
@@ -0,0 +1,317 @@
+        assert_eq!(
+            statistics[1].column_statistics[0].max_value,
+            Precision::Exact(ScalarValue::Int32(Some(2)))
+        );
+        assert_eq!(
+            statistics[1].column_statistics[0].min_value,
+            Precision::Exact(ScalarValue::Int32(Some(1)))
+        );
+    }
+
+    #[tokio::test]
+    async fn test_statistics_by_partition_of_data_source() -> datafusion_common::Result<()>
+    {
+        let scan = generate_listing_table_with_statistics(Some(2)).await;
+        let statistics = scan.statistics_by_partition()?;
+        // Check the statistics of each partition
+        assert_eq!(statistics.len(), 2);
+        for stat in &statistics {
+            assert_eq!(stat.column_statistics.len(), 2);
+            assert_eq!(stat.total_byte_size, Precision::Exact(110));
+        }
+        check_unchanged_statistics(statistics);
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_statistics_by_partition_of_projection() -> datafusion_common::Result<()>
+    {
+        let scan = generate_listing_table_with_statistics(Some(2)).await;
+        // Add projection execution plan
+        let exprs: Vec<(Arc<dyn PhysicalExpr>, String)> =
+            vec![(Arc::new(Column::new("id", 0)), "id".to_string())];
+        let projection = ProjectionExec::try_new(exprs, scan)?;
+        let statistics = projection.statistics_by_partition()?;
+        for stat in &statistics {
+            assert_eq!(stat.column_statistics.len(), 1);
+            assert_eq!(stat.total_byte_size, Precision::Exact(8));
+        }
+        check_unchanged_statistics(statistics);
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_statistics_by_partition_of_sort() -> datafusion_common::Result<()> {
+        let scan = generate_listing_table_with_statistics(Some(2)).await;
+        // Add sort execution plan
+        let sort = SortExec::new(
+            LexOrdering::new(vec![PhysicalSortExpr {
+                expr: Arc::new(Column::new("id", 0)),
+                options: SortOptions {
+                    descending: false,
+                    nulls_first: false,
+                },
+            }]),
+            scan,
+        );
+        let mut sort_exec = Arc::new(sort.clone());
+        let statistics = sort_exec.statistics_by_partition()?;
+        assert_eq!(statistics.len(), 1);
+        assert_eq!(statistics[0].num_rows, Precision::Exact(4));
+        assert_eq!(statistics[0].column_statistics.len(), 2);
+        assert_eq!(statistics[0].total_byte_size, Precision::Exact(220));
+        assert_eq!(
+            statistics[0].column_statistics[0].null_count,
+            Precision::Exact(0)
+        );
+        assert_eq!(
+            statistics[0].column_statistics[0].max_value,
+            Precision::Exact(ScalarValue::Int32(Some(4)))
+        );
+        assert_eq!(
+            statistics[0].column_statistics[0].min_value,
+            Precision::Exact(ScalarValue::Int32(Some(1)))
+        );
```

Comment:
I found it hard to read here what the expected statistics were

What do you think about a pattern like this (to create the expected statistcs)

```rust
```suggestion
        let expected_statistics = Statistics {
          num_rows: Precision::Exact(4),
          total_bute_size:Precision::Exact(220),
          column_statistics: vec![
            ColumnStatistics {...
            }]
          };
        assert_eq!(statistics, expected_statistics);
```

Reply:
Thanks for the suggestion, I believe after https://github.com/apache/datafusion/pull/15503/commits/667337b60b718a3faa74f0fbcfc616abb0880176,  it'll be clear.